src/overlays/actors/ovl_En_Yb/z_en_yb.c

```diff
@@ -37,59 +44,443 @@ const ActorInit En_Yb_InitVars = {
+                    func_80BFA2FC(globalCtx);
+
+                } else {
+                    func_801477B4(globalCtx);
+                    this->actionFunc = EnYb_ReceiveMask;
+                    EnYb_ReceiveMask(this, globalCtx);
+                }
+                break;
+            default:
+                func_801477B4(globalCtx);
+                this->actionFunc = EnYb_Idle;
+                break;
+        }
+    }
+    EnYb_EnableProximityMusic(this);
+}
+
+void EnYb_TeachingDanceFinish(EnYb* this, GlobalContext* globalCtx) {
+    EnYb_UpdateAnimation(this, globalCtx);
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        this->actionFunc = EnYb_Talk;
+        // Spread my dance across the world
+        func_801518B0(globalCtx, 0x147C, &this->actor);
+        this->actor.flags &= ~ACTOR_FLAG_10000;
+    } else {
+        func_800B8500(&this->actor, globalCtx, 1000.0f, 1000.0f, -1);
+    }
+    EnYb_EnableProximityMusic(this);
+}
+
+// dancing countdown
+void EnYb_TeachingDance(EnYb* this, GlobalContext* globalCtx) {
+    EnYb_UpdateAnimation(this, globalCtx);
+
+    if (this->teachingCutsceneTimer > 0) {
+        this->teachingCutsceneTimer--;
+    } else {
+        EnYb_FinishTeachingCutscene(this);
+        this->actionFunc = EnYb_TeachingDanceFinish;
+        this->actor.flags |= ACTOR_FLAG_10000;
+        func_800B8500(&this->actor, globalCtx, 1000.0f, 1000.0f, -1);
+    }
+    EnYb_EnableProximityMusic(this);
+}
+
+void EnYb_Idle(EnYb* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Player* player = GET_PLAYER(globalCtx);
+
+    EnYb_UpdateAnimation(this, globalCtx);
+    if (this->actor.xzDistToPlayer < 180.0f && fabsf(this->actor.playerHeightRel) < 50.0f &&
+        globalCtx->msgCtx.ocarinaMode == 3 && globalCtx->msgCtx.unk1202E == 7 &&
+        gSaveContext.playerForm == PLAYER_FORM_HUMAN) {
+        this->actionFunc = EnYb_TeachingDance;
+        this->teachingCutsceneTimer = 200;
+        EnYb_ChangeCutscene(this, 0);
+    } else if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        func_80BFA2FC(globalCtx);
+        this->actionFunc = EnYb_Talk;
+        if (Player_GetMask(globalCtx) == PLAYER_MASK_KAMARO) {
+            // I have taught you, go use it
+            func_801518B0(globalCtx, 0x147C, &this->actor);
+        } else {
+            // regular talk to him first dialogue
+            func_801518B0(globalCtx, 0x147B, &this->actor);
+        }
+    } else if (EnYb_CanTalk(this, globalCtx)) {
+        func_800B8614(&this->actor, globalCtx, 120.0f);
+    }
+
+    if (this->unkFlag410 & 1) {
+        if (!(player->stateFlags2 & 0x8000000)) {
+            this->unkFlag410 &= ~1;
+        }
+    } else if ((player->stateFlags2 & 0x8000000) && this->actor.xzDistToPlayer < 180.0f &&
+               fabsf(this->actor.playerHeightRel) < 50.0f) {
+
+        this->unkFlag410 |= 1;
+        Actor_PlaySfxAtPos(&this->actor, NA_SE_SY_TRE_BOX_APPEAR);
+    }
```

Comment:
Is this for when you're playing the ocarina to him or something?

Reply:
Just checked, yes, it goes off when you first pull out your ocarina standing in front of him, before you start playing any notes